tests/integration/api/mutations/addCartItems/addCartItems.test.js

```diff
@@ -0,0 +1,119 @@
+import Factory from "/imports/test-utils/helpers/factory";
+import TestApp from "/imports/test-utils/helpers/TestApp";
+import { encodeCartOpaqueId, encodeCartItemsOpaqueIds } from "@reactioncommerce/reaction-graphql-xforms/cart";
+import { createFactoryForSchema } from "@reactioncommerce/data-factory";
+import AddCartItemsMutation from "./AddCartItemsMutation.graphql";
+import Logger from "@reactioncommerce/logger";
+import SimpleSchema from "simpl-schema";
+
+let testApp;
+let addCartItems;
+let fakeCart;
+let fakeCartItem;
+let shopId;
+let originalError;
+let cartItemInputSchemaFactory;
+
+
+
+beforeAll(async () => {
+  testApp = new TestApp();
+  await testApp.start();
+  shopId = await testApp.insertPrimaryShop();
+  addCartItems = testApp.mutate(AddCartItemsMutation);
+  cartItemInputSchemaFactory = createFactoryForSchema(
```

Comment:
@aldeed 
 i'm not sure how else to accomplish this I need the input schema for the CartItems for this test so i can mock a correct order item submission.

is this accessible in the `testApp`?

Reply:
which comes from [this](https://github.com/reactioncommerce/reaction/blob/develop/imports/plugins/core/cart/server/no-meteor/util/addCartItems.js) file...

Reply:
You could code the whole object and not use a factory. It's a fairly small object. It also could be fine to export the schema from the plugin for test factory purposes. We don't have a great pattern for this yet.